### DISCUSSION TITLE
#2 📑 Weekly Report 2

### DISCUSSION THREAD

**Discussion by ghost** [SE-TINF22B6/Statify#26] [Blog]
Hello everybody👋
The second week is almost over, so we want to give you an update on our project.

## UML Diagram
This week we mainly dealt with software requirement analysis.
To understand the main functions our application should have, we have drawn a UML diagram:
![UML_Use_Cases](https://github.com/SE-TINF22B6/Statify/assets/64202248/fbfad1dd-06c3-4bfd-ba4b-a9d201fb0a59)

## UI Design
Based on the main use cases shown in the UML diagram, we created our first UI mockups in Figma. As our project is based on the Spotify API, the UI design uses "Spotify Green" as one of the main colors. 
You can look at our UI mockups in Figma [here](https://www.figma.com/file/JuoCjSig1km8syVQRFGuFX?node-id=0:1&locale=en&type=design).

![IMG_0768](https://github.com/SE-TINF22B6/Statify/assets/64202248/3617f9a4-e08d-458f-b50b-ac76b84e33c3)
![IMG_0769](https://github.com/SE-TINF22B6/Statify/assets/64202248/84243700-ca1d-47ab-ab8e-1110295b5644)
![IMG_0770](https://github.com/SE-TINF22B6/Statify/assets/64202248/c39a50aa-d4df-4c41-a9c7-0ff831b3878c)
![IMG_0771](https://github.com/SE-TINF22B6/Statify/assets/64202248/4c194a97-294e-486d-90b9-46a236e27969)
![IMG_0772](https://github.com/SE-TINF22B6/Statify/assets/64202248/e3f6d38f-ec45-48a0-95b4-28a9c34df22d)

## Software Requirement Specifications
To learn more about the different use cases we developed with the help of the UML diagram, you can check out our [Software Requirement Specifications](https://github.com/SE-TINF22B6/Statify/blob/main/Documentation/srs.md).

## Backend
We also set up the backend of our project for which we will use Spring Boot. Now that everything is set up, we can soon start with the "real" development and programming.

Stay tuned for more to come!

**Comment by derrrobin** [comment]
Hey Tthere,
since you haven't uploaded a report for this week yet, I'll give you my feedback on this one.

This one is pretty positive.

I really like the first design you uploaded. I also like that you gave us a glimpse of how you are working on your project and what design you chose. I also like that you have already presented us different use cases like logging in or registering or even the general use of your application. 

The design is very much to my taste. It's playful and colorful but still clean and not too cluttered. You have really found a good balance.

I am really looking forward to seeing your progress.

Your CookPit Team

Translated with www.DeepL.com/Translator (free version)

**Comment by JonasEbbinghaus25** [comment]
Hi, 
your weekly blog is very detailed. Also the UML diagram is designed very detailed. However, it is a bit small and when you click on the diagram to view it larger, the labels of the respective ellipses disappear. 

Your mockups look very appealing and also the chosen colors fit super together. However, some components overlap in the images. 

Also your Software Requirement Specifications are very detailed and the used UML digrams illustrate the processes within your application.

**Comment by sfrancisw** [comment]
Hello there!
The UI Mockups look very promising and good. I'm interested to see how well and easy these mockups will transform into code. 
Your idea is also very interesting and we are looking forward to seeing where your project is heading.
The UML diagram is very clearly structured.

Hope to see more progress soon!

Team time2shine
